### PR TITLE
add tests for most layer types

### PR DESCRIPTION
**Pull request by thanushipeiris**
Two very simple tests for the shapes layer (Issue #48)

Test 1: reference image layer, moving shapes layer
Test 2: reference shapes layer, moving shapes layer

### CODE REVIEW

**Comment by jni** [comment]
Thanks @thanushipeiris!

I think for this task what we actually need is to produce reference layers of *each* type corresponding to the cameraman and rotated cameraman (as in the existing image<>image test), then use [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/how-to/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) on each layer type pair to make sure things work. That means that only a *single* test needs to get written, but it tests all possible layer combinations. Does that make sense? Let me know if you need help with this!

For the reference shape, points, and vectors data, you can draw them in napari on top of the cameraman image and then create sample data in text. For labels, you can e.g. manually paint the cameraman, then save the data as a .zarr array somewhere in the test directory. zarr compresses labels very effectively so it's a good way to generate the data.

An alternative is to try to get a reasonable segmentation of the cameraman programmatically using watershed with specific markers, and filters.farid.

**Comment by thanushipeiris** [comment]
@jni I think I've done that for Image, Shapes, Points and Labels layers but have yet to add Vector and Tracks layers (also should I include tracks layers??).

I was using the camera and rotated camera as the reference and moving layers for all layer types btw.

File: src/affinder/_tests/test_affinder.py [hunk 1]
```diff
@@ -2,36 +2,80 @@
...
-                                                 [314.75696913, 375.13825634],
-                                                 [184.33085012, 439.81718637]])
-    actual = np.asarray(l1.affine)
-    expected = np.array(  # yapf: ignore
-            [[0.50221294, 0.86131375, 3.38128256],
-             [-0.86478707, 0.50303866, 324.04591946],
-             [0., 0., 1.]])
-
-    np.testing.assert_allclose(actual, expected)
+        viewer=viewer,
```
Comment (jni) [inline src/affinder/_tests/test_affinder.py:65]: Make sure you format with yapf

**Comment by jni** [comment]
Great! You're right, for tracks you will need to add mixed-dimensionality alignment, so let's leave that layer out for now!

Could you also rechunk the zarr files so that they are one chunk? ie set chunk size to 512x512.

**Comment by jni** [comment]
Thanks @thanushipeiris! The CI isn't passing. You can see from [this line](https://github.com/jni/affinder/runs/5087485763?check_suite_focus=true#step:7:41) in the failing build that this is because zarr is not installed in the test environment, which can be fixed by adding zarr to [this section](https://github.com/jni/affinder/blob/fe73724cf1eb899b885d3199c8f5f0099576b150/setup.cfg#L49) of setup.cfg.

**Comment by thanushipeiris** [comment]
I added zarr as a dependency to `tox.ini` as well and less tests are failing, but some are still failing and I'm not sure how to fix it.

**Comment by jni** [comment]
Ah, damn, yeah this one is tricky, the *only* reason I know about it is that it happened on napari a few days ago and Talley fixed it! LOL Here's the references:

- https://github.com/GabrielBB/xvfb-action/issues/25
- https://github.com/napari/napari/pull/4046

Can you look at napari/napari#4046 and imitate that in our test running? Thanks!

**Comment by thanushipeiris** [comment]
It seems to work now! I didn't need to switch to [tlambert03/setup-qt-libs@v1](https://github.com/napari/napari/pull/4046/files#diff-150360a98d1e3a2914cfa79bdb95df88169a57574de35fc0af8e611121fa8c15R69) for the Linux tests, although I don't really understand what's happening here so let me know if I should do that change too.

**Comment by thanushipeiris** [comment]
Alright so I read a little bit more and I believe it's just a set of libraries that are needed for qt packaged together in the one repo. It's tangential to fixing the CI issue but more compact so I put it in anyways.

**Comment by jni** [comment]
Whoooooo! :tada: Thanks for pushing this all the way over the finish line, @thanushipeiris! :smiley: :tada: :rocket:

**Pull request merged**